decisions/0004-parallel-path.md

```diff
@@ -0,0 +1,137 @@
+# Print Formatting
+
+- Status: **Proposed**
+- Deciders: Adam Washington, Dan Nixon, Tristan Youngs
+- Date: 16-11-2020
+
+## Context and Problem Statement
+
+The need for a comprehensive parallelism solution throughout the code
+to enable large simulations to be completed in a reasonable time while
+running on diverse architectures.
+
+
+## Decision Drivers
+
+- Current MPI solution does not easily support multiple platforms
+- Need to support multiple hardware architectures (e.g. multi-core, GPU)
+
+
+## Considered Options
+
+1. [MPI](https://www.mpi-forum.org)
+2. [Parallel STL](https://en.cppreference.com/w/cpp/header/execution)
+3. [OpenMP](https://openmp.org)
+4. [CUDA](https://developer.nvidia.com/cuda-zone)
+5. [OpenCL](https://www.khronos.org/opencl/)
+6. [CyCL](https://www.khronos.org/sycl/)
+
+## Decision Outcome
+
+We ultimately have a 4 stage path forward
+
+### 1) Modernisation
+
+We continue moving the code base toward standard C++ containers.  This
+provides more information to the compiler and should provide
+performance benefits under all circumstances.  Even if the roadmap for
+parallelism changes, it is likely that this will still provide benefits
+and highly unlikely that it will be a hindrance.
+
+### 2) STL Parallelism
+
+Using standard containers will enable us to use standard algorithms,
+many of which now support parallelism as of C++17.  This will involve
+converting many of our loops into algorithms and lambdas.  Beyond
+immediately providing a speed-up on multi-core machines, this process
+moves the code base into a position that will be easier to parallelize
+through one of the GPU libraries
+
+### 3) CyCL
```

Comment:
```suggestion
### 3) SYCL
```

Reply:
And throughout the following paragraphs.